Kernel/FileSystem/FATFileSystem.cpp

```diff
@@ -0,0 +1,361 @@
+        } else if (entry->attributes == 0x0F) {
+            dbgln_if(FAT_VERY_DEBUG, "FATFS: Found LFN entry");
+            lfn_entries.append(*(FATLongFileNameEntry*)entry);
+        } else {
+            dbgln_if(FAT_VERY_DEBUG, "FATFS: Found 8.3 entry");
+            lfn_entries.reverse();
+            auto inode = FATInode::create(fs(), *entry, lfn_entries);
+            inodes.append(inode);
+            lfn_entries.clear();
+        }
+    }
+
+    return EINVAL;
+}
+
+void FATInode::compute_filename(Vector<FATLongFileNameEntry> lfn_entries)
+{
+    VERIFY(!m_filename);
+
+    // FIXME: Find a way to handle errors here instead of using MUST
+    if (lfn_entries.is_empty()) {
+        StringBuilder filename;
+        filename.append(byte_terminated_string(m_entry.filename, 8, ' '));
+        if (m_entry.extension[0] != ' ') {
+            filename.append('.');
+            filename.append(byte_terminated_string(m_entry.extension, 3, ' '));
+        }
+        m_filename = MUST(KString::try_create(filename.string_view()));
+    } else {
+        StringBuilder filename;
+        for (auto& lfn_entry : lfn_entries) {
+
+            filename.append(lfn_entry.characters1[0]);
+            filename.append(lfn_entry.characters1[1]);
+            filename.append(lfn_entry.characters1[2]);
+            filename.append(lfn_entry.characters1[3]);
+            filename.append(lfn_entry.characters1[4]);
+            filename.append(lfn_entry.characters2[0]);
+            filename.append(lfn_entry.characters2[1]);
+            filename.append(lfn_entry.characters2[2]);
+            filename.append(lfn_entry.characters2[3]);
+            filename.append(lfn_entry.characters2[4]);
+            filename.append(lfn_entry.characters2[5]);
+            filename.append(lfn_entry.characters3[0]);
+            filename.append(lfn_entry.characters3[1]);
+        }
+
+        m_filename = MUST(KString::try_create(byte_terminated_string(filename.string_view().bytes().data(), filename.length(), 0xFF)));
+    }
+
+    dbgln_if(FAT_VERY_DEBUG, "FATFS: Computed filename {} for inode {}", m_filename, index());
+}
+
+time_t FATInode::fat_date_time(u16 date, u16 time) const
+{
+    time_t year = (date >> 9) + (1980 - 1970);
+    time_t month = (date >> 5) & 0b1111;
+    time_t day = date & 0b11111;
+
+    time_t hour = time >> 11;
+    time_t minute = (time >> 5) & 0b1111111;
+    time_t second = (time & 0b11111) << 1;
+
+    return second + minute * 60 + hour * 3600
+        + day + (24 * 3600) + month * (30 * 24 * 3600) + year * (365 * 24 * 3600);
+}
+
+// TODO: This might be worth putting as a part of the StringView class
+StringView FATInode::byte_terminated_string(u8 const* string, size_t max_length, u8 byte) const
+{
+    size_t length = max_length;
+
+    // HACK: size_t is unsigned, so we can't check if it's smaller than 0,
+    //       but when we go below zero the value turns really big
+    while (length <= max_length) {
```

Comment:
That sounds like a prime opportunity for some weird values causing issues down the line (not that we probably survive until here anyways if something passes a length of `-1`).

Couldn't you start out with putting the `string` and `max_length` into a `StringView` at the very beginning, and then continue with cutting off bytes at the end? That would seem like a much safer alternative compared to working with raw pointers and offsets.